dataikuapi/dss/plugin.py

```diff
@@ -147,4 +175,148 @@ def put_file(self, path, f):
         data = f.read() # eat it all, because making it work with a path variable and a MultifilePart in swing looks complicated
         return self.client._perform_empty("POST", "/plugins/%s/contents/%s" % (self.plugin_id, path), raw_body=data)
 
-        
\ No newline at end of file
+
+class DSSPluginUsage(object):
+    """
+    Information on a usage of an element of a plugin.
+    """
+    def __init__(self, json_object):
+        self._json_object = json_object
+
+    def get_raw(self):
+        return self._json_object
+
+    @property
+    def element_kind(self):
+        """
+        Element kind (webapps, python-formats,...)
+        :return: the element kind
+        :rtype: str
+        """
+        return self._json_object["elementKind"]
+
+    @property
+    def element_type(self):
+        """
+        Element type
+        :return: the element type
+        :rtype: str
+        """
+        return self._json_object["elementType"]
+
+    @property
+    def object_id(self):
+        """
+        :return: Id of the object using the plugin element
+        :rtype: str or none
+        """
+        return self._json_object.get("objectId", None)
+
+    @property
+    def object_type(self):
+        """
+        :return: Type of the object using the plugin element
+        :rtype: str or none
+        """
+        return self._json_object.get("objectType", None)
+
+    @property
+    def project_key(self):
+        """
+        :return: Project key of the object using the plugin element
+        :rtype: str or none
+        """
+        return self._json_object.get("projectKey", None)
+
+
+class DSSMissingType(object):
+    """
+    Information on a type not found while analyzing usages of a plugin.
+    """
+    def __init__(self, json_object):
+        self._json_object = json_object
+
+    def get_raw(self):
+        return self._json_object
+
+    @property
+    def missing_type(self):
+        """
+        :return: the missing type
+        :rtype: str
+        """
+        return self._json_object["missingType"]
+
+    @property
+    def object_id(self):
+        """
+        :return: Id of the object relying on the missing type
```

Comment:
Not sure I understand "relying on"?

Reply:
Replaced by "depending". Hope it is clearer.